PMSCore/ProjectClass.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace PMSCore {
+    //class Task with attributes: Name, Description, Status; private attribute Status demonstrates encapsulation
+    public class Task { 
+        public string Name { get; set; } //naming the Task
+        public string Description { get; set; } //providing a brief description of the Task
+        public string Status { get; private set; } = "ToDo"; //initial Status is ToDo, it is protected (only accessible from within the class and derived classes)
+
+        public Task(string name, string description) //constructor part, to initialize a new task, hiding internal components of the Task => abstraction   
+        {
+            Name = name; //setting the name for it
+            Description = description; //providing a description
+        }
+
+        public void UpdateStatus(string newStatus) //method that updates a new value for this task => abstraction (status can't be changed directly)
+        {
+            if (newStatus == "ToDo" || newStatus == "In Progress" || newStatus == "Completed") //update status only if it matches one of these: "ToDo", "In Progress", "Completed"
+                Status = newStatus;
+        }
+    }
```

Comment:
This is not part of this ticket nor this PR there is work in progress for the TaskClass, if you need it you should create an empty one